src/components/organisms/Common/AssetContainer/hooks.ts

```diff
@@ -74,7 +71,21 @@ export default (teamId?: string, initialAssetUrl?: string | null, allowDeletion?
     data?.assets.pageInfo?.hasNextPage || data?.assets.pageInfo?.hasPreviousPage;
 
   const isRefetching = networkStatus === 3;
-  const assets = data?.assets.edges?.map(e => e.node) as AssetNodes;
+
+  const assets = data?.assets.edges
+    .map<Asset | undefined>(asset =>
+      asset.node
+        ? {
+            id: asset.node.id,
+            name: asset.node.name,
+            size: asset.node.size,
+            teamId: asset.node.teamId,
+            url: asset.node.url,
+            contentType: asset.node.contentType,
+          }
+        : undefined,
+    )
+    .filter((asset): asset is Asset => !!asset);
```

Comment:
I think you can keep it more concise if you do something like this:
```
const assets = data?.assets.edges
    .map<Asset | undefined>(({ node })=> node)
    .filter((asset): asset is Asset => !!asset);
```

Please test to make sure it works though.
Also, what is the purpose of the filter? Is there any chance that there'll be an undefined asset inside the assets array?